File: Modules/EBanking/Accounts/AccountsForm.cs
using cyberbanking.EBanking.Accounts;
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace cyberbanking.EBanking.Forms;

[FormScript("EBanking.Accounts")]
[BasedOnRow(typeof(AccountsRow), CheckNames = true)]
public class AccountsForm
{
    [Hidden]
    public Boolean IsActive { get; set; }

    public decimal Balance { get; set; }


    public AccountType AccountType { get; set; }

    [HideOnInsert]
    public DateTime OpenDate { get; set; }

    [HideOnInsert]
    public int CustomerId { get; set; }
}

File: Modules/ServerTypes/EBanking/AccountsForm.ts
import { BooleanEditor, DecimalEditor, EnumEditor, DateEditor, LookupEditor, PrefixedContext } from "@serenity-is/corelib";
import { AccountType } from "./Accounts.AccountType";
import { initFormType } from "@serenity-is/corelib/q";

export interface AccountsForm {
    IsActive: BooleanEditor;
    Balance: DecimalEditor;
    AccountType: EnumEditor;
    OpenDate: DateEditor;
    CustomerId: LookupEditor;
}

export class AccountsForm extends PrefixedContext {
    static formKey = 'EBanking.Accounts';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!AccountsForm.init)  {
            AccountsForm.init = true;

            var w0 = BooleanEditor;
            var w1 = DecimalEditor;
            var w2 = EnumEditor;
            var w3 = DateEditor;
            var w4 = LookupEditor;

            initFormType(AccountsForm, [
                'IsActive', w0,
                'Balance', w1,
                'AccountType', w2,
                'OpenDate', w3,
                'CustomerId', w4
            ]);
        }
    }
}

[AccountType]; // referenced types